cmd/kubeadm/app/util/config/initconfiguration.go

```diff
@@ -251,47 +256,47 @@ func DefaultedInitConfiguration(versionedInitCfg *kubeadmapiv1.InitConfiguration
 }
 
 // LoadInitConfigurationFromFile loads a supported versioned InitConfiguration from a file, converts it into internal config, defaults it and verifies it.
-func LoadInitConfigurationFromFile(cfgPath string) (*kubeadmapi.InitConfiguration, error) {
+func LoadInitConfigurationFromFile(cfgPath string, skipCRIDetect bool) (*kubeadmapi.InitConfiguration, error) {
```

Comment:
> there's no other way to know when to skip it, we have to add the extra argument everywhere?

The tricky part is here.
It's hard to know if we can skip the CRI detection without an additional parameter or context for `LoadOrDefaultInitConfiguration` -> `LoadInitConfigurationFromFile`.

Any suggestions?

Reply:
When we spread things everywhere, it tastes bad. However, we have to do that sometimes.

In other cases, they may have several things that need to be options/flags everywhere. They use a `xxxOptions` struct with several bools or some other type attributes. Then later, if we have further logic to add to spread everywhere, we can add a new option in this options struct then. But this looks like an over-design if we only have one flag here.

Reply:
> When we spread things everywhere, it tastes bad. However, we have to do that sometimes.
> 
> In other cases, they may have several things that need to be options/flags everywhere. They use a `xxxOptions` struct with several bools or some other type attributes. Then later, if we have further logic to add to spread everywhere, we can add a new option in this options struct then. But this looks like an over-design if we only have one flag here.

an options struct or a variadic arg list sounds like something we can do at some point.

we can log a k/kubeadm issue for it.

Reply:
An example here : https://github.com/kubernetes/kubernetes/blob/534a0941e8d39926015dccfaab0e37251a070cf7/pkg/apis/core/validation/validation.go#L3814-L3830

Reply:
imo, we can do with with the extra argument that we pass on many places, the way this PR is.
but we should log a ticket in k/kubeadm for refactor with struct ^.

Reply:
kubeadm ticket: https://kugithub.com/kubernetes/kubeadm/issues/2941

Reply:
/lgtm
I'm sorry for not getting back to you sooner after the holiday.
Let's follow it at https://github.com/kubernetes/kubeadm/issues/2941.